### PR TITLE
fix duplicate code in netty-example module.

### PR DESCRIPTION
**Pull request by coding2world**
Motivation:

I found some duplicated code  when reading the netty code.

Modification:

all the changes in the  netty-example module.

Result:

fix duplicate code in netty-example module.

### CODE REVIEW

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package io.netty.example.util;
+
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+public class ServerUtil {
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/util/ServerUtil.java:9]:
```suggestion
public final class ServerUtil {
```

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(ServerUtil.class);
+
+    private ServerUtil() {
+    }
+
+    static final boolean SSL = System.getProperty("ssl") != null;
+
+    public static SslContext getSslContext() {
+
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/util/ServerUtil.java:19]:
```suggestion
```

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        try {
+            SelfSignedCertificate ssc = new SelfSignedCertificate();
+            sslCtx = SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey())
+                    .build();
+        } catch (Exception ex) {
+            logger.warn("exception occurs when get SslContext.", ex);
+        }
+        return sslCtx;
+    }
+
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/util/ServerUtil.java:34]:
```suggestion
```

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+public class ServerUtil {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(ServerUtil.class);
+
+    private ServerUtil() {
+    }
+
+    static final boolean SSL = System.getProperty("ssl") != null;
+
+    public static SslContext getSslContext() {
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/util/ServerUtil.java:18]: I think we should rename this to `buildServerSslContext`

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package io.netty.example.util;
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/util/ServerUtil.java:1]: missing license header

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: example/src/main/java/io/netty/example/util/ServerUtil.java (hunk 4 above)
Comment (coding2world) [reply example/src/main/java/io/netty/example/util/ServerUtil.java:18]: Thanks for your suggestion.

File: example/src/main/java/io/netty/example/util/ServerUtil.java (hunk 5 above)
Comment (coding2world) [reply example/src/main/java/io/netty/example/util/ServerUtil.java:1]: add a file named package-info.java?

**Comment by coding2world** [comment]
@normanmaurer 
Hi, i have changed my code.  You can review my code when you are free. Thank you !

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2012 The Netty Project
```
Comment (chrisvest) [inline example/src/main/java/io/netty/example/util/ServerUtil.java:2]: Update copyright year to 2022.

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            return null;
+        }
+
+        SslContext sslCtx = null;
+        try {
+            SelfSignedCertificate ssc = new SelfSignedCertificate();
+            sslCtx = SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey())
+                    .build();
+        } catch (Exception ex) {
+            logger.warn("exception occurs when get SslContext.", ex);
```
Comment (chrisvest) [inline example/src/main/java/io/netty/example/util/ServerUtil.java:47]: Just let the exception bubble out like the examples did previously. No need for any logging as the stack trace will be printed when the exception unwinds `main()`.

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: example/src/main/java/io/netty/example/util/ServerUtil.java (hunk 6 above)
Comment (coding2world) [reply example/src/main/java/io/netty/example/util/ServerUtil.java]: Thanks for your suggestion.

File: example/src/main/java/io/netty/example/util/ServerUtil.java (hunk 7 above)
Comment (coding2world) [reply example/src/main/java/io/netty/example/util/ServerUtil.java]: Thanks for your suggestion.

File: example/src/main/java/io/netty/example/util/ServerUtil.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        if (!SSL) {
+            return null;
+        }
+
+        SslContext sslCtx = null;
+        try {
+            SelfSignedCertificate ssc = new SelfSignedCertificate();
+            sslCtx = SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey())
+                    .build();
+        } catch (Exception ex) {
```
[the remark this answers is not in the document]
Comment (coding2world) [reply example/src/main/java/io/netty/example/util/ServerUtil.java]: Thanks for your suggestion.

**Review by chrisvest — APPROVED** [review]

**Comment by coding2world** [comment]
@normanmaurer 
Hi, i have changed my code. You can review my code when you are free. Thank you !

**Comment by normanmaurer** [comment]
@coding2world can you please sign our icla and let me know once done: https://netty.io/s/icla ?

**Comment by coding2world** [comment]
> coding2world

**Comment by coding2world** [comment]
> @coding2world can you please sign our icla and let me know once done: https://netty.io/s/icla ?

@normanmaurer  Hi, I haved signed your icla.

**Comment by coding2world** [comment]
@normanmaurer Hi, I have signed your icla. please check this pr when you are free.

**Comment by chrisvest** [comment]
@coding2world There's a really weird build error. Can you try doing a rebase on the latest 4.1 branch and see if that helps?

**Comment by coding2world** [comment]
> @coding2world There's a really weird build error. Can you try doing a rebase on the latest 4.1 branch and see if that helps?

👌

**Comment by coding2world** [comment]
@chrisvest Hi, I have rebase the code. Could you help me  approval the workflow.

**Comment by chrisvest** [comment]
@coding2world Thanks!

**Pull request merged**